Doc/c-api/object.rst

```diff
@@ -613,3 +613,19 @@ Object Protocol
 
    .. versionadded:: 3.14
 
+.. c:function:: int PyUnstable_IsImmortal(PyObject *obj)
+
+   This function returns ``1`` if *obj* is :term:`immortal`, and ``0`` otherwise. This function cannot fail.
+
+   Immortal objects don't care about reference counting, thus they no-op calls to :c:func:`Py_INCREF`
```

Comment:
I don't think that it's the right place to elaborate on immortal objects: the glossary would be a better place. You already linked to the "immortal" term. Maybe delete this paragraph?